src/app/client/src/app/modules/public/module/player/components/public-collection-player/public-collection-player.component.html

```diff
@@ -2,57 +2,67 @@
   <div class="ui stackable grid">
 
     <!--Content Section-->
-    <div class="twelve wide column" [appTelemetryImpression]="telemetryContentImpression" id="app-player-collection-renderer" *ngIf="showPlayer">
+    <div class="twelve wide column" [appTelemetryImpression]="telemetryContentImpression"
+      id="app-player-collection-renderer" *ngIf="showPlayer && playerContent">
 
       <!--Content Title-->
       <div class="sb-pageSection">
         <div class="sb-pageSection-header">
           <h1 class="sb-pageSection-title m-0 mr-8">
-            {{contentTitle}}
+            {{playerContent.name}}
           </h1>
-          <button 
-            class="sb-btn sb-btn-xs sb-btn-error ml-auto" 
-            type="button" 
-            (click)="closeContentPlayer()" 
-            appTelemetryInteract 
-            [telemetryInteractCdata]="this.telemetryCdata" 
-            [telemetryInteractObject]="playerTelemetryInteractObject" 
-            [telemetryInteractEdata]="closePlayerInteractEdata"
-            > 
+          <div class="ml-auto">
+            <span *ngIf="isOffline" class="mr-16">
+              <button (click)="downloadContent(playerContent)"
+                *ngIf="checkOfflineRoutes === 'browse' && playerContent?.addedToLibrary !== true && playerContent?.showAddingToLibraryButton !== true"
```

Comment:
can move all this login in ts file and have one property to show hide this button

Reply:
we need to have telemetry event for this